### ISSUE TITLE
Worker Service queue_delay directive not working for me

### ISSUE THREAD

**Issue by ivan94** [aws/copilot-cli#3016]
I tried creating a Worker Service and setup Autoscaling using the queue_delay directive. The worker is a long-running task that can take on average 10 minutes to complete, so I set that as the msg_processing_time, and as a test I set acceptable_latency 30m which should result. That should make the service scale up after 3 messages are queued.

I then queued 6 different messages and observed the service but no other tasks were provisioned. It continued with only 1 task running all the way until all 6 messages were processed.

Some additional information that may or may not be relevant:

Currently, this Worker Service is the only service in the copilot application. The messages are sent to the worker queue from an outside application. That also means I'm not currently using the SNS topic subscription part of the Worker Service, since I had no other service set up as a publisher. I'm not sure if the autoscaling relies on any of that.

I initially set the minimum instances to 0, as I wanted to test if I was able to reduce costs by not running any workers when the queue was empty. This resulted in no task being provisioned ever. Thinking it could be a glitch with the 0 value, I changed that to 1, which created 1 task, but it never scaled up past that.

Here is the current manifest configuration:

``` YAML
# Your service name will be used in naming your resources like log groups, ECS services, etc.
name: job-service
type: Worker Service

# Configuration for your containers and service.
image:
  # Docker build arguments.
  build: Dockerfile

cpu: 2048       # Number of CPU units for the task.
memory: 10240    # Amount of memory in MiB used by the task.
# exec: true     # Enable running commands in your container.

subscribe:
  queue:
    timeout: 25m
    dead_letter:
      tries: 4

count:
  range:
    min: 1
    max: 50
    spot_from: 26
  queue_delay:
    acceptable_latency: 30m
    msg_processing_time: 10m```

**Comment by efekarakus** [comment]
Hi @ivan94  👋 

You're right I could also reproduce the issue, thanks for bringing this issue to our attention!

### Root cause

The issue seems to be because we wrote our template assuming that there will be topic subscriptions required to setup autoscaling:
https://github.com/aws/copilot-cli/blob/2a08f939fc9f45e01ad25d24b37568c6fe8c6234/internal/pkg/template/templates/workloads/partials/cf/autoscaling.yml#L84

We should be able to relax this restriction and just have:
```
{{- if .Autoscaling.QueueDelay }}
```

### Mitigation
In the mean time, until we send a fix for it you can get around this issue by updating your manifest like this:
```yaml
subscribe:
  topics: # Trick copilot into assuming there are registered topics
  -
  queue:
    timeout: 25m
    dead_letter:
      tries: 4
```

I deployed with the manifest change above and autoscaling was picked up.

Hope this helps!

**Comment by ivan94** [comment]
I just tried your mitigation strategy and it worked for me. Thank you so much!

**Comment by efekarakus** [comment]
Awesome! Glad it worked 😄 We'll keep the issue open until we have a fix in place

**Comment by efekarakus** [comment]
This is now released 🎉 in v1.13.0 https://github.com/aws/copilot-cli/releases/tag/v1.13.0 !
